crossword-xml-uploader/src/main/scala/com/gu/crossword/crosswords/ComposerCrosswordIntegration.scala

```diff
@@ -1,30 +1,35 @@
 package com.gu.crossword.crosswords
 
+import java.nio.ByteBuffer
+
+import com.amazonaws.handlers.AsyncHandler
 import com.gu.crossword.Config
 import com.gu.crossword.crosswords.models.CrosswordXmlFile
-import com.gu.crossword.services.Http.httpClient
-import com.squareup.okhttp._
+import com.gu.crossword.services.Kinesis
+import com.amazonaws.services.kinesis.model.{ PutRecordsRequestEntry, PutRecordsRequest, PutRecordsResult }
+
 import scala.xml._
 
-trait ComposerCrosswordIntegration extends XmlProcessor with CrosswordStore {
+trait ComposerCrosswordIntegration extends Kinesis {
 
   def createPage(crosswordXmlFile: CrosswordXmlFile, crosswordXmlToCreatePage: Elem)(implicit config: Config): Unit = {
-    val request = buildRequest(crosswordXmlToCreatePage)
-    val response: Response = httpClient.newCall(request).execute()
-    if (response.isSuccessful)
-      println(s"Successfully created page for crossword ${crosswordXmlFile.key}. ${response.body.string}")
-    else
-      println(s"Creating page for crossword ${crosswordXmlFile.key} failed with error: ${response.body.string}")
 
-  }
+    val record = new PutRecordsRequestEntry()
+      .withPartitionKey(crosswordXmlFile.key)
+      .withData(ByteBuffer.wrap(crosswordXmlToCreatePage.toString.getBytes))
+
+    val request = new PutRecordsRequest()
+      .withStreamName(config.composerCrosswordIntegrationStreamName)
+      .withRecords(record)
+
+    kinesisClient.putRecordsAsync(request, new AsyncHandler[PutRecordsRequest, PutRecordsResult] {
```

Comment:
Given the number of crosswords per day, probably not, but I agree with you, it would be better done synchronously.